Source/HedgeGI/ModelProcessor.cpp

```diff
@@ -143,14 +145,18 @@ void ModelProcessor::processGenerationsStage(const std::string& directoryPath, P
             auto name = toUtf8(entry.name());
 
             Logger::logFormatted(LogType::Normal, "Loading %s...", name.data());
-            
+
             hl::archive archive = CabinetCompression::load(entry.file_data(), entry.size());
```

Comment:
This will not work if the level is already compressed. You can change this to use ArchiveCompression instead, it checks for the signature and uses the appropriate decompressor.